sdk/dotnet/Pulumi/Core/InputUnion.cs

```diff
@@ -20,12 +21,18 @@ private InputUnion(Output<Union<T0, T1>> oneOf)
 
```

Comment:
Note the two constructors above strict me as weird.
Why on earth do we allow `new InputUnion()` to make an empty union?! That looks like a great source for bugs.
Two why is the constructor from `Output<Union>` private? We've got a conversion operator for it now but maybe that should just be made public aswell/instead.